### DISCUSSION TITLE
Libcurl 7.83.1 Build failed: Can't open libcrypto.lib

### DISCUSSION THREAD

**Discussion by Sgharat** [curl/curl#8980] [Q&A]
I was trying to upgrade libcurl 7.83.1 for avoiding security vulnerability in my current project.
But Unlike 7.82, I am not able to build libcurl 7.83.1.It is giving below error in my environment.

LINK : fatal error LNK1181: cannot open input file 'libcrypto.lib'.

Can anyone confirm , how to fix this issue?
Is there any dependencies of openssl 1.1.1 to build libcurl 7.83.1?

**Comment by captain-caveman2k** [comment]
I am seeing it as well when building with the Visual Studio projects and using OpenSSL. Are you performing the same or using the winbuild or cmake build mechanisms, and did you build OpenSSL?

I believe the problem I see is related to an inconsistency between how the Visual C compiler version number is being used in the directory path when building OpenSSL (and presumably for WolfSSL as well) via our batch files and that used for the linker path in the Visual Studio projects.

I tried to build the "DLL Debug - OpenSSL DLL" configuration for x64 using Visual Studio 2022 where the libcurl linker path is defined as:

`..\..\..\..\..\openssl\build\Win64\VC14.30\DLL Debug;%(AdditionalLibraryDirectories)`

However, using build-openssl.bat the resulting binaries are placed in:

`D:\curl\openssl\build\Win64\VC14.3\DLL Debug`

I also note that:

- generate.bat uses vc14.10 and vc14.30 as arguments to specify Visual Studio 2017 and 2022
- build-openssl.bat and build-wolfssl use vc14.1, vc14.2 and vc14.3 to specify Visual Studio 2017, 2019 and 2022

The following fixes the build problem when using OpenSSL - but it doesn't address the inconsistency I mentioned above with the batch file arguments (That should be addressed as a separate issue IMHO):

```
diff --git a/projects/build-openssl.bat b/projects/build-openssl.bat
index 6feac9428..d095d9c48 100644
--- a/projects/build-openssl.bat
+++ b/projects/build-openssl.bat
@@ -73,7 +73,7 @@ rem ***************************************************************************
       set "VC_PATH=Microsoft Visual Studio 14.0\VC"
     ) else if /i "%~1" == "vc14.1" (
       set VC_VER=14.1
-      set VC_DESC=VC14.1
+      set VC_DESC=VC14.10

       rem Determine the VC14.1 path based on the installed edition in descending
       rem order (Enterprise, then Professional and finally Community)
@@ -86,7 +86,7 @@ rem ***************************************************************************
       )
     ) else if /i "%~1" == "vc14.2" (
       set VC_VER=14.2
-      set VC_DESC=VC14.2
+      set VC_DESC=VC14.20

       rem Determine the VC14.2 path based on the installed edition in descending
       rem order (Enterprise, then Professional and finally Community)
@@ -99,7 +99,7 @@ rem ***************************************************************************
       )
     ) else if /i "%~1" == "vc14.3" (
       set VC_VER=14.3
-      set VC_DESC=VC14.3
+      set VC_DESC=VC14.30

       rem Determine the VC14.3 path based on the installed edition in descending
       rem order (Enterprise, then Professional and finally Community)
```

And also when using WolfSSL:

```
diff --git a/projects/build-wolfssl.bat b/projects/build-wolfssl.bat
index 7a871e75e..54bbdfd83 100644
--- a/projects/build-wolfssl.bat
+++ b/projects/build-wolfssl.bat
@@ -76,7 +76,7 @@ rem ***************************************************************************
     set "VC_PATH=Microsoft Visual Studio 14.0\VC"
   ) else if /i "%~1" == "vc14.1" (
     set VC_VER=14.1
-    set VC_DESC=VC14.1
+    set VC_DESC=VC14.10
     set VC_TOOLSET=v141

     rem Determine the VC14.1 path based on the installed edition in descending
@@ -90,7 +90,7 @@ rem ***************************************************************************
     )
   ) else if /i "%~1" == "vc14.2" (
     set VC_VER=14.2
-    set VC_DESC=VC14.2
+    set VC_DESC=VC14.20
     set VC_TOOLSET=v142

     rem Determine the VC14.2 path based on the installed edition in descending
@@ -104,7 +104,7 @@ rem ***************************************************************************
     )
   ) else if /i "%~1" == "vc14.3" (
     set VC_VER=14.3
-    set VC_DESC=VC14.3
+    set VC_DESC=VC14.30
     set VC_TOOLSET=v143

     rem Determine the VC14.3 path based on the installed edition in descending

```
